### PR TITLE
[spark] Support hilbert curve sort action in spark

### PR DESCRIPTION
**Pull request by TaoZex**
### Purpose

**Support hilbert curve sort action**

Relying on this repository：https://github.com/davidmoten/hilbert-curve. It is Apache-2.0 and is compatible with the incubator-paimon project licensing.

Linked issue: (https://github.com/apache/incubator-paimon/issues/1412)

### API and Format

no

### CODE REVIEW

**Comment by TaoZex** [comment]
Hello, this is the first time for me to make contribution for paimon. If you have spare time, please kindly help to review this pr, and I will modify the code according to your suggestions. Thank you. @leaves12138 @JingsongLi

File: paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/utils/ConvertBinaryUtil.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.paimon.spark.utils;
+
+import java.nio.charset.StandardCharsets;
+
+/** BinaryUtil util to convert the value to binary. */
+public class ConvertBinaryUtil {
+    private ConvertBinaryUtil() {}
```
Comment (leaves12138) [inline paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/utils/ConvertBinaryUtil.java:25]: Can't you add a “/n” between `public class ConvertBinaryUtil` and `private ConvertBinaryUtils(){}`

File: paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/sort/SparkHilbertUDF.java [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+    private static final int BITS_NUM = 63;
+
+    SparkHilbertUDF() {}
+
+    private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundException {
+        in.defaultReadObject();
+    }
+
+    byte[] hilbertCurvePosBytes(Seq<Long> points) {
+        List<Long> longs = (List<Long>) JavaConverters.seqAsJavaList(points);
```
Comment (leaves12138) [inline paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/sort/SparkHilbertUDF.java:65]: I think (List<Long>) could be dropped. Idea tells that Casting 'JavaConverters.seqAsJavaList(...)' to 'List<Long>' is redundant

**Review by leaves12138 — COMMENTED** [review]
Thanks for @TaoZex, it's a great job for hilbert curve support, just two small tips left.

**Comment by leaves12138** [comment]
By the way, if you can add some tests for `ConvertBinaryUtil.java`, that would be wonderful!

**Comment by TaoZex** [comment]
> Thanks for @TaoZex, it's a great job for hilbert curve support, just two small tips left.

Thank you for your suggestions. I have modified the code according to your suggestions. By the way, have a great weekend!

**Review by leaves12138 — APPROVED** [review]
+1

**Pull request merged**